srcDoc/examples.rst

```diff
@@ -600,3 +600,359 @@ Maximum Throughput -- 2 *Routing Service*
+    The function takes three arguments:
+
+        - A reference to the customer type data.
+        - A specific number unique for every key.
+        - The target size set by the command line parameter ``-dataLen <bytes>``
+        minus the overhead of RTI Perftest. If applicable, the user can use this
+        value to set the content of the data.
+
+::
+
+    bool set_custom_type_data(
+            RTI_CUSTOM_TYPE & data,
+            unsigned long key,
+            int target_data_len)
+    {
+        bool success = true;
+        data.test_long = key;
+        if (sprintf(data.test_string.test_string, "Hello World! %lu", key) < 0) {
+            success = false;
+        }
+        return success;
+    }
+
+- **finalize_custom_type_data**:
+    This function is used to remove your data. It is called in the destructor.
+    The function takes one argument:
+
+        - A reference to the customer type data.
+
+::
+
+    bool finalize_custom_type_data(RTI_CUSTOM_TYPE & data)
+    {
+        return true;
+    }
+
+- **initialize_custom_type_dynamic_data**:
+    This function is used to initialize your DynamicData.
+    The user will be able to allocate memory or set an immutable field of the data.
+    The function takes one argument:
+
+        - A reference to the full DDS_DynamicData object including custom_type.
+
+::
+
+    bool initialize_custom_type_dynamic_data(DDS_DynamicData & data)
+    {
+        bool success = long_seq.maximum(0);
+        if (!success) {
+            fprintf(stderr, "long_seq.maximum failed.\n");
+        }
+        return success;
+    }
+
+- **register_custom_type_dynamic_data**:
+    This function is used to set your DynamicData before been register. It is
+    only required for key types.
+    The user will set the key field of the data, base on the key input.
+    There is a one to one mapping between an input key and an instance.
+    The function takes two arguments:
+
+        - A reference to the full DDS_DynamicData object including custom_type.
+        - A specific number unique for every key.
+
+::
+
+    void register_custom_type_dynamic_data(DDS_DynamicData & data, unsigned long key)
+    {
+        // TODO initialize DDS_DynamicData object to be registered
+        DDS_ReturnCode_t retcode = data.set_long(
+                "custom_type.test_long",
+                DDS_DYNAMIC_DATA_MEMBER_ID_UNSPECIFIED,
+                key);
+        if (retcode != DDS_RETCODE_OK) {
+            fprintf(stderr, "set_long(test_long) failed: %d.\n", retcode);
+        }
+    }
+
+- **set_custom_type_dynamic_data**:
+    This function is used to set your DynamicData before been sent.
+    It is called everytime that data is sent. It is the responsibility of the
+    user to set the custom type data before been sent with the right "key"
+    value and the "target_data_len".
+    The function takes three arguments:
+
+        - A reference to the full DDS_DynamicData object including custom_type.
```

Comment:
either "object that includes the custom_type" or "object, including the custom_type," whichever you mean